### PR TITLE
HPCC-21440 Remove audiences and message classes that will no longer be used

### PR DESCRIPTION
**Pull request by shamser**
…e used

Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Review by shamser — COMMENTED** [review]
@ghalliday

File: system/jlib/jscm.hpp [hunk 1]
```diff
@@ -183,12 +183,8 @@ class StringBuffer;
 
 typedef enum
 {
-    MSGAUD_unknown     = 0x00,
     MSGAUD_operator    = 0x01,
```
Comment (ghalliday) [inline system/jlib/jscm.hpp]: These could do with some comments

File: system/jlib/jlog.hpp [hunk 2]
```diff
@@ -100,14 +88,12 @@ inline const char * LogMsgAudienceToFixString(LogMsgAudience audience)
 
 typedef enum
 {
-    MSGCLS_unknown     = 0x00,
     MSGCLS_disaster    = 0x01,
```
Comment (ghalliday) [inline system/jlib/jlog.hpp]: These could do with some comments highlighting the differences

File: system/jlib/jlog.hpp [hunk 3]
```diff
@@ -127,10 +113,6 @@ inline const char * LogMsgClassToVarString(LogMsgClass msgClass)
         return("Information");
     case MSGCLS_progress:
         return("Progress");
-    case MSGCLS_legacy:
```
Comment (ghalliday) [inline system/jlib/jlog.hpp]: legacy hasn't been removed yet - here and below

File: system/jlib/jlog.hpp [hunk 4]
```diff
@@ -54,18 +54,12 @@ inline const char * LogMsgAudienceToVarString(LogMsgAudience audience)
...
-        return("Performance");
-    case MSGAUD_internal:
-        return("Internal");
     case MSGAUD_programmer:
         return("Programmer");
     case MSGAUD_legacy:
         return("Legacy");
     case MSGAUD_audit:
-        return("Audit");
+       return("Audit");
```
Comment (ghalliday) [inline system/jlib/jlog.hpp]: indent: space removed by mistake

File: system/jlib/jlog.cpp [hunk 5]
```diff
@@ -2185,16 +2185,16 @@ void attachManyLogMsgMonitorsFromPTree(IPropertyTree * tree)
...
 const LogMsgCategory MCoperatorWarning(MSGAUD_operator, MSGCLS_warning);
-const LogMsgCategory MCinternalWarning((LogMsgAudience)(MSGAUD_internal & MSGAUD_programmer), MSGCLS_warning, 1);
+const LogMsgCategory MCinternalWarning(MSGAUD_programmer, MSGCLS_warning, 1);
 const LogMsgCategory MCuserProgress(MSGAUD_user, MSGCLS_progress);
 const LogMsgCategory MCoperatorProgress(MSGAUD_operator, MSGCLS_progress);
 const LogMsgCategory MCdebugProgress(MSGAUD_programmer, MSGCLS_progress);
 const LogMsgCategory MCdebugInfo(MSGAUD_programmer, MSGCLS_information);
-const LogMsgCategory MCstats(MSGAUD_performance, MSGCLS_information);
-const LogMsgCategory MCevent(MSGAUD_monitor, MSGCLS_event);
+const LogMsgCategory MCstats(MSGAUD_programmer, MSGCLS_information);
```
Comment (ghalliday) [inline system/jlib/jlog.cpp]: From your jira comments I think this should be MSGCLS_progress.  Do we need to distinguish progress from status?  I suspect not.

File: system/jlib/jlog.hpp (hunk 2 above)
Comment (ghalliday) [reply system/jlib/jlog.hpp]: E.g. Especially what is the difference between information and progress

File: system/jlib/jlog.cpp [hunk 6]
```diff
@@ -2185,16 +2185,16 @@ void attachManyLogMsgMonitorsFromPTree(IPropertyTree * tree)
...
+const LogMsgCategory MCevent(MSGAUD_operator, MSGCLS_information);
```
Comment (ghalliday) [inline system/jlib/jlog.cpp]: Is MCevent still a usefulname?  Also should it be progress or information?

**Review by ghalliday — COMMENTED** [review]
@richardkchapman please look at the roxie changes
@shamser a few comments.

File: system/jlib/jlog.hpp (hunk 3 above)
Comment (shamser) [reply system/jlib/jlog.hpp]: @ghalliday I'll be removing legacy in a separate ticket (https://track.hpccsystems.com/browse/HPCC-21442).  I'll mention that in this ticket.

File: system/jlib/jlog.cpp (hunk 6 above)
Comment (shamser) [reply system/jlib/jlog.cpp]: Agree. I'll change this.

File: system/jlib/jlog.hpp (hunk 3 above)
Comment (ghalliday) [reply system/jlib/jlog.hpp]: which is why this code should not be removed in this PR...

**Comment by ghalliday** [comment]
@shamser

File: system/jlib/jlog.hpp (hunk 3 above)
Comment (shamser) [reply system/jlib/jlog.hpp]: oops

**Comment by shamser** [comment]
@ghalliday

**Comment by ghalliday** [comment]
Looks good

**Comment by shamser** [comment]
@ghalliday

**Pull request merged**